lib/object_space_profile.rb

```diff
@@ -0,0 +1,55 @@
+# ObjectSpaceProfile provides a tool for assessing how many Ruby objects are in
+# memory. By taking measures over time you can watch the change in the number of
+# objects, and identify situations in which we're holding on to more objects
+# than intended. Sometimes it may be necessary to generate a report with each
+# cycle, especially if performance means you aren't getting to the end of the
+# process.
```

Comment:
Not quite sure what you mean by `Sometimes it may be necessary to generate a report with each
# cycle, especially if performance means you aren't getting to the end of the
# process.`

Do you mean, necessary to generate the output of this profiler for one iteration of the loop in `study.each_well_for_qc_report_in_batches` (qc_report.rb)?

Reply:
Yes, I had to spit one out in each iteration, as it was dying before it even got past the third loop through. (Although this was partly because the initial version of the profiler was interacting very badly with configatron, and spawned new objects when it attempted to interrogate the settings there).